### ISSUE TITLE
Data Science FAQ: univariate interactions and ridge feature selection

### ISSUE THREAD

**Issue by kalelooz** [g-leech/argmin-gravitas#2373]
Two statements in the feature-selection section of [Data Science FAQ](https://www.gleech.org/data-science/) need correcting.

**1. Univariate selection does not account for interactions between the input features.**

The accuracy discussion says univariate methods work well “since they factor in interactions between features”. They score each input separately against the outcome, so they can miss a relationship that only appears when inputs are considered together.

For example, let two independent binary inputs be equally likely to take each of these four combinations:

| First input | Second input | Outcome: inputs differ |
| --- | --- | --- |
| 0 | 0 | 0 |
| 0 | 1 | 1 |
| 1 | 0 | 1 |
| 1 | 1 | 0 |

Each input alone has zero information about the outcome; together they determine it exactly. This is the XOR example discussed in section 3.3 and Figure 3 of [Guyon and Elisseeff (2003)](https://www.jmlr.org/papers/volume3/guyon03a/guyon03a.pdf), which the FAQ already cites. The later paragraph on filters also correctly notes this limitation.

Suggested replacement:

> Univariate feature selection scores each feature separately. It can miss useful combinations, so interactions may require constructed features or a method that evaluates features together.

**2. Regularisation does not necessarily remove coefficients; ridge is an example.**

The embedding paragraph equates lower complexity with “that is, fewer coefficients”, then lists ridge alongside lasso and elastic net. Ridge can reduce model complexity by shrinking coefficients while keeping every feature. It does not, by itself, perform feature selection.

For a two-column identity input matrix, an outcome vector of `[1, 2]`, no intercept and a ridge penalty of 1, the fitted coefficients are `[0.5, 1.0]`. Both are still nonzero. More generally, this example gives `[1, 2] / (1 + alpha)` for any finite nonnegative penalty.

This follows directly from the [ridge objective](https://scikit-learn.org/stable/modules/linear_model.html#ridge-regression-and-classification). The [feature-selection guide](https://scikit-learn.org/stable/modules/feature_selection.html#l1-based-feature-selection) explains why penalties that produce sparse coefficients, such as lasso, can serve as feature selectors.

Suggested replacement:

> Regularisation reduces model complexity. Some penalties, such as lasso's L1 penalty, can set coefficients to zero and select features. Ridge shrinks coefficients but generally keeps them nonzero.

**Runnable check**

The following passed with NumPy 2.3.5 and scikit-learn 1.7.2. It checks all four XOR inputs, rather than a random sample, and compares ridge's fitted coefficients with the exact solution.

```python
import numpy as np
from sklearn.feature_selection import mutual_info_classif
from sklearn.linear_model import Ridge

X = np.array([[0, 0], [0, 1], [1, 0], [1, 1]])
y = np.array([0, 1, 1, 0])
individual_bits = mutual_info_classif(X, y, discrete_features=True) / np.log(2)
joint_feature = (2 * X[:, 0] + X[:, 1]).reshape(-1, 1)
joint_bits = mutual_info_classif(joint_feature, y, discrete_features=True) / np.log(2)
np.testing.assert_allclose(individual_bits, [0, 0], atol=1e-12)
np.testing.assert_allclose(joint_bits, [1], atol=1e-12)

for alpha in (0.0, 1.0, 100.0):
    weights = Ridge(alpha=alpha, fit_intercept=False).fit(np.eye(2), [1, 2]).coef_
    np.testing.assert_allclose(weights, np.array([1, 2]) / (1 + alpha))
    assert np.count_nonzero(weights) == 2

print("PASS")
```

Could you consider these two corrections under your [error-bounty policy](https://www.gleech.org/bounty)?

**Comment by g-leech** [comment]
I'm classifying this as a nitpick, which entitles you to a thankyou. Thank you!
